.github/workflows/code-style-lint.yml

```diff
@@ -0,0 +1,36 @@
+# Workflow that runs prettier code style check.
+name: Code Style and Lint
+
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+
+jobs:
+  prettier:
+    name: Style Check
+    if: ${{ (github.repository_owner != 'aws-solutions' && github.event_name == 'push') || (github.repository_owner == 'aws-solutions' && github.event_name == 'pull_request') }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
+    steps:
+      - uses: actions/checkout@v3
+      - run: npx --y prettier --config source/.prettierrc.yml --check 'source/**/*.ts'
+  linter:
+    name: Lint Check
+    if: ${{ (github.repository_owner != 'aws-solutions' && github.event_name == 'push') || (github.repository_owner == 'aws-solutions' && github.event_name == 'pull_request') }}
```

Comment:
doesn't this mean that the check will not be run if we push to a feature branch that is not on a fork?

Reply:
Correct, but it will be run when the PR is opened to merge the feature branch.